### DISCUSSION TITLE
Dynamically Detect Variable Annotations

### DISCUSSION THREAD

**Discussion by EricScottN** [koxudaxi/datamodel-code-generator#2155] [Q&A]
Not sure this exists as a flag or combination of flags, but I was wondering if there was a way to automatically detect duplicate class variables and extract/replace them with variable annotations at the top of the module. For example, assume the following two classes were created from a json file:

```python
class ClassA(BaseModel):
    abbrev: str
    score: int
    place_name: Annotated[str, Field(alias='placeName')]

class ClassB(BaseModel):
    id: int
    market: str
    place_name: Annotated[str, Field(alias='placeName')]
```

What I would like to see instead:
```python

PlaceName = Annotated[str, Field(alias='placeName')]

class ClassA(BaseModel):
    abbrev: str
    score: int
    place_name: PlaceName

class ClassB(BaseModel):
    id: int
    market: str
    place_name: PlaceName
```

**Comment by koxudaxi** [comment]
I updated the ducumentation. Thank you .
https://github.com/koxudaxi/datamodel-code-generator/pull/2896
